### ISSUE TITLE
http basic auth passwords not redacted in all locations for CLI output.

### ISSUE THREAD

**Issue by joekir** [cloudfoundry/bosh-cli#223]
## Description

Upon receiving an agent timeout while attempting to deploy a BOSH director via [this guide](https://bosh.io/docs/init-google.html) I received the following error:     

```
Deploying:
  Creating instance 'bosh/0':
    Waiting until instance is ready:
      Sending ping to the agent:
        Performing request to agent endpoint 'https://mbus:953pei2eiaicffwfer@10.0.0.6:6868/agent':
          Performing POST request:
            Post https://mbus:<redacted>@10.128.0.6:6868/agent: dial tcp 10.0.0.6:6868: i/o timeout

Exit code 1
```

*Note: I changed the real password value between `mbus:` and `@` but couldn't use `<redacted>` as that is what the actual CLI output should be :)*

## Code reference

https://github.com/cloudfoundry/bosh-cli/blob/47ea06453290186de2610150498d3901d16c4131/vendor/github.com/cloudfoundry/bosh-agent/agentclient/http/agent_request.go#L38

**Comment by cppforlife** [comment]
ah seems like we didnt bump agentclient in the vendor directory. story: https://www.pivotaltracker.com/story/show/145948135

**Comment by cppforlife** [comment]
should be fixed in 2.0.23 (https://bosh.io/docs/cli-v2.html#install)
